### PR TITLE
JAMES-3140 Injections for CachedBlobStore (finally)

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/modules/blobstore/BlobStoreConfigurationTest.java [hunk 1]
```diff
@@ -223,4 +223,33 @@ void fromShouldReturnConfigurationWhenBlobStoreImplIsSupportedAndHasExtraSpaces(
                 .getName())
             .isEqualTo(CASSANDRA);
     }
+
+    @Test
+    void cacheEnabledShouldBeTrueWhenSpecified() {
```
Comment (Arsnael) [inline server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/modules/blobstore/BlobStoreConfigurationTest.java:228]: `cacheEnabled` or `cacheEnable`? as the conf parameter is `cache.enable` (applies for tests after too if agreed)

File: src/site/xdoc/server/config-blobstore.xml [hunk 2]
```diff
@@ -49,6 +49,35 @@
                 <dd>hybrid: Using both objectstorage for unfrequently read or big blobs &amp; cassandra for small, often read blobs</dd>
             </dl>
 
+
+            <subsection name="Cassandra BlobStore Cache">
+                <p>A cassandra cache can be enabled to reduce latency when reading small blobs frequently.
```
Comment (Arsnael) [inline src/site/xdoc/server/config-blobstore.xml:54]:
```suggestion
                <p>A Cassandra cache can be enabled to reduce latency when reading small blobs frequently.
```

File: server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/modules/blobstore/BlobStoreConfigurationTest.java (hunk 1 above)
Comment (chibenwa) [reply server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/modules/blobstore/BlobStoreConfigurationTest.java:228]: But he getter is named `cacheEnabled` and this is the property tested in the left part of the assertion.

File: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties [hunk 3]
```diff
@@ -12,6 +12,29 @@ implementation=objectstorage
 # Optional, defaults to 32768 bytes (32KB), must be positive
 hybrid.size.threshold=32768
 
+# ========================================= Cassandra BlobStore Cache ======================================
+# A cassandra cache can be enabled to reduce latency when reading small blobs frequently
+# A dedicated keyspace with a replication factor of one is then used
+# Cached eviction policy is TTL based
+# Only blobs below a given threshold will be stored.
```
Comment (nvduc91) [inline dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties:19]: Did we do `<=` threadHoldSize, `below` should be `with in`

File: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties [hunk 4]
```diff
@@ -12,6 +12,29 @@ implementation=objectstorage
...
+# To be noted that blobs are stored within a single Cassandra row, hence a low threshold should be used.
+
+# Enable the cache? Optional and default to false. Must be a boolean.
+cache.enable=false
+
+# Cached eviction policy is TTL based. Optional and defaults to 7 days. Must be a duration.
```
Comment (nvduc91) [inline dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties:25]: s/Cached/Cache

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blob.properties [hunk 5]
```diff
@@ -6,6 +6,28 @@
 # hybrid is using both objectstorage for unfrequently read or big blobs & cassandra for small, often read blobs
 implementation=objectstorage
 
+# ========================================= Cassandra BlobStore Cache ======================================
+# A cassandra cache can be enabled to reduce latency when reading small blobs frequently
+# A dedicated keyspace with a replication factor of one is then used
+# Cached eviction policy is TTL based
```
Comment (nvduc91) [inline dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blob.properties:12]: s/Cached/Cache

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blob.properties [hunk 6]
```diff
@@ -6,6 +6,28 @@
...
+# Only blobs below a given threshold will be stored.
+# To be noted that blobs are stored within a single Cassandra row, hence a low threshold should be used.
+
+# Enable the cache? Optional and default to false. Must be a boolean.
+cache.enable=false
+
+# Cached eviction policy is TTL based. Optional and defaults to 7 days. Must be a duration.
```
Comment (nvduc91) [inline dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blob.properties:19]: idem

File: server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java [hunk 7]
```diff
@@ -96,25 +114,33 @@ static BlobStoreConfiguration from(Configuration configuration) {
...
 
-        return new BlobStoreConfiguration(blobStoreImplName);
+        boolean cacheEnabled = configuration.getBoolean(CACHE_ENABLE_PROPERTY, false);
+
+        return new BlobStoreConfiguration(blobStoreImplName, cacheEnabled);
     }
 
     public static BlobStoreConfiguration cassandra() {
-        return new BlobStoreConfiguration(BlobStoreImplName.CASSANDRA);
+        return new BlobStoreConfiguration(BlobStoreImplName.CASSANDRA, !CACHE_ENABLED);
```
Comment (nvduc91) [inline server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java:123]: Did the context clear enough to import static?

Comment (chibenwa) [reply server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java:123]: The constant is defined in this file.

File: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties (hunk 3 above)
Comment (chibenwa) [reply dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties:19]: IMO `within` should be used with a range, and a range cannot be represented as a upper bound, even if it is inclusive.

I also don't think that the upper bound being inclusive makes much of a difference, so I rather keep it as is.

File: server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java (hunk 7 above)
Comment (nvduc91) [reply server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/blobstore/BlobStoreConfiguration.java:123]: ok i agree

**Comment by chibenwa** [comment]
I fixed the Cached -> cache typo upon merge

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**